src/main/java/com/google/sps/data/NSWGovAdaptor.java

```diff
@@ -32,6 +32,18 @@ public NSWGovAdaptor() throws Exception {
     return this.dataMap;
   }
 
+  /**
+   * Manages the entire process of getting data from the NSW Gov API,
+   * from fetching the data to collating it into the unified AQDataPoint format.
+   * @return the AQI data for the current date with duplicate elements removed
+   * @throws Exception
+   */
+  public ArrayList<AQDataPoint> getAQIData() throws Exception {
+    LocalDate currentDate = LocalDate.now();
+    ArrayList<AQDataPoint> dataWithDupes = convertAllDataPoints(extractAQI(currentDate));
```

Comment:
```suggestion
    ArrayList<AQDataPoint> dataWithDupes = this.convertAllDataPoints(extractAQI(currentDate));
```
?

Reply:
What does the `this` change? My understanding was that it would just call the method on the instance for non-static objects?

Reply:
In the specific context, it doesn't **change** anything; however, it's making the implicit clear.

It's only strictly required when attempting to access attributes shadowed by parameters, and a few special cases. For example:
```
public class Rectangle {
    private int x, y;
    private int width, height;
        
    public Rectangle() {
        this(0, 0, 1, 1);
    }
    public Rectangle(int width, int height) {
        this(0, 0, width, height);
    }
    public Rectangle(int x, int y, int width, int height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }
}
```
Or to allow call chaining:
```
public class ButWhy {
    private int a, b, c;

    public ButWhy WithA(int** a) {
        this.a = a;
        return this;
    }

    public ButWhy WithB(int b) {
        this.b = b;
        return this;
    }

    public ButWhy WithC(int c) {
        this.c = c;
        return this;
    }

    public void DoSomething() {
        int answer = this.a + this.b + this.c;
        System.out.println("Hah! No! ("+answer+")");
    }
}
...
ButWhy noReason = new ButWhy();
noReason().WithA(11).WithB(17).WithC(14).DoSomething();
...
```

However, using `this` where it **could** be used, in general practice, removes any potential ambiguity.

Reply:
Ok, that makes sense. I've added the `this` in the places you suggested